src/utils/Processing.ts

```diff
@@ -355,6 +356,36 @@ export function writeFHIRResources(
   logger.info(`Exported ${count} FHIR resources as JSON.`);
 }
 
+export function writePreprocessedFSH(outDir: string, inDir: string, tank: FSHTank) {
+  const preprocessedPath = path.join(outDir, '_preprocessed');
+  fs.ensureDirSync(preprocessedPath);
+  // Because this is the FSH that exists after processing, some entities from the original FSH are gone.
+  // Specifically, Aliases have already been resolved to their corresponding values, and the insertion
+  // of RuleSets have already been applied.
+  // TODO: Add Resources and Logicals once they are being imported and stored in docs
+  tank.docs.forEach(doc => {
+    let fileContent = '';
+    const entities = [
+      ...doc.profiles.values(),
+      ...doc.extensions.values(),
+      ...doc.instances.values(),
+      ...doc.valueSets.values(),
+      ...doc.codeSystems.values(),
+      ...doc.invariants.values(),
+      ...doc.mappings.values()
+    ];
+    // Sort entities by original line number, then write them out.
+    sortBy(entities, 'sourceInfo.location.startLine').forEach(entity => {
+      fileContent += `// Originally defined on lines ${entity.sourceInfo.location.startLine} - ${entity.sourceInfo.location.endLine}${EOL}`;
+      fileContent += `${entity.toFSH()}${EOL}${EOL}`;
+    });
+    const outPath = path.relative(inDir, doc.file);
+    fs.ensureFileSync(path.join(preprocessedPath, outPath));
+    fs.writeFileSync(path.join(preprocessedPath, outPath), fileContent);
```

Comment:
I think before writing the file, we want to check that it actually has content. Something like a file of only aliases will end up empty after preprocessing. Then we can either not write this file at all, or write a comment like: 
```
// This file has no content after preprocessing
```
Personally I think I prefer the comment for transparency.

Reply:
Actually, I think we may need to make some changes to how aliases are handled in general. The one issue is that the way we use aliases, they are not actually all resolved in the importer. There are cases like this:
```
* extension[USCoreRace].extension[ombCategory].valueCoding = OmbRaceCat#2106-3 "White"
```
where `USCoreRace` is an alias, but we don't resolve that in the importer, since the importer just treats the entire path as a string, but rather when we go to `fish` for the `USCoreRace` definition. I'm not totally sure how you want to handle this. The options I see are:
1. Do still output aliases in the preprocessed output
2. Somehow figure out a way to resolve these aliases in the rules we are `toFSH`ing
I think 2 seems harder than 1, but I'm not totally sure. Alternatively there is an option 3 which would be that we don't worry too much about this, but on the other hand someone looking at this output could get confused wondering what `USCoreRace` points to, since that alias doesn't exist, and nothing in the dependencies or FSH is defined with that name. What do you think?

Reply:
I'm not seeing any cases like this in the most recent version on the `fsh-mcode` repo, and I don't entirely understand what you're trying to explain here. Are there any other examples you can give me of what you're talking about?

Reply:
I think if you check out the https://github.com/HL7/fhir-mCODE-ig repo, you should be able to see an example of this. On [this](https://github.com/HL7/fhir-mCODE-ig/blob/c2e1fdcaee8e1734720b7213b1d936318090a1be/input/fsh/EX_ExtendedExample.fsh#L743) line, `USCoreRace` is used as an alias, defined in the `AL_USCore.fsh` file. If you check out SUSHI, in `FSHTank.ts`'s `fish` function, we do this:
```
    item = this.resolveAlias(item) ?? item;
```
and that is the line that resolves the alias, it doesn't happen on the import side.

Reply:
Ah, that clears things up! So, since we're not entirely done with Aliases after these preprocessing steps, I think it makes sense to include them in the output. The fishing is happening in `setPropertyOnInstance`, which doesn't get the rule itself, just a `PathPart[]` that we've built. The FSH output by this feature is meant to represent the objects we have in memory, and the Rule's path does not change as a result of this Alias resolution.

Fortunately, a file with nothing but RuleSets should still be empty after preprocessing, so that will be useful for demonstration in a test case.

Reply:
Ok great, makes sense to me.